src/main/java/app/model/boundary/BoundaryImp.java

```diff
@@ -56,4 +63,19 @@ public Vector getTeleport()
     {
         return null;
     }
+
+    public boolean isSolid()
```

Comment:
I thought the point of the enum was to get rid of all of these boolean flags and methods.  Like having a `SOLID_TRANSPARENT` enum and `SOLID_VISIBLE` to make up the 9 possible options within the enum and swich on that, so I don't really understand why you have all three methods still (as only the `isSolid()` is used)

Reply:
they are used in GraphicsEngine line 61 and 71.
But I agree they can be removed and use it on the type itself. I debated over it, but since it's cleaner when using the method, I kept them in. I'll remove them though.